src/journal/ObjectRecorder.cc

```diff
@@ -266,18 +162,14 @@ void ObjectRecorder::handle_append_flushed(uint64_t tid, int r) {
     m_in_flight_tids.erase(tid_iter);
 
     InFlightAppends::iterator iter = m_in_flight_appends.find(tid);
-    if (r == -EOVERFLOW || m_overflowed) {
-      if (iter != m_in_flight_appends.end()) {
-        ldout(m_cct, 10) << "append overflowed" << dendl;
-        m_overflowed = true;
-      } else {
-        // must have seen an overflow on a previous append op
-        ceph_assert(r == -EOVERFLOW && m_overflowed);
-      }
+    ceph_assert(iter != m_in_flight_appends.end());
+
+    if (r == -EOVERFLOW) {
+      ldout(m_cct, 10) << "append overflowed" << dendl;
+      m_overflowed = true;
 
       // notify of overflow once all in-flight ops are complete
-      if (m_in_flight_tids.empty() && !m_aio_scheduled) {
-        m_append_buffers.splice(m_append_buffers.begin(), m_pending_buffers);
+      if (m_in_flight_tids.empty()) {
```

Comment:
Just thinking, is a scenario possible, when we send two appends (1) and (2); for (1) the osd returns 0, for (2) the osd returns -EOVERFLOW, but (2) completes earlier on the client side? Or there are some guarantee about the order?

Reply:
Ops are ordered on the wire (both directions) and librados uses a single thread for all callbacks, so order is guaranteed.